### PR TITLE
Fast Track: apply verdicts fail closed

### PR DESCRIPTION
**Pull request by michal-lightly**
## What has changed and why?

Adds the Fast Track bot orchestrator (`run-bot.ts`): derive the trusted PR target from GitHub state, resolve the effective verdict via `verdict-policy`, and either approve a current pass or dismiss the bot approval for every other outcome. The pass path re-checks the target after approving and dismisses if the PR changed during the write.

Stacked on the verdict-policy PR (#1690), which is now based on `main`. Retarget this PR to `main` once #1690 merges.

## How has it been tested?

`make -C fast_track static-checks` and `make -C fast_track test` pass. `run-bot.test.ts` covers approve+comment on a pass, dismissal on a failure verdict, the current-opt-out-label override, the head-changed-during-run race, and the no-eligible-target skip. Verdict-decision edge cases live in `verdict-policy.test.ts` (#1690).

## Did you update CHANGELOG.md?

- [ ] Yes
- [x] Not needed (internal change)

## Summary by CodeRabbit

* **New Features**
  * Added automated bot handling for eligible pull requests based on evaluation verdicts.
  * Approves passing pull requests and updates their status comments.
  * Dismisses approvals for non-passing verdicts, including cases where the PR is opted out.
  * Skips processing when no eligible pull request matches.
* **Bug Fixes**
  * Prevents stale approvals by rechecking the PR state after approval and dismissing if it no longer matches.
  * Dismisses approvals when post-approval refresh/recheck fails or returns no matching target.
* **Tests**
  * Added a comprehensive test suite covering approval, dismissal, skipping, and refresh edge cases.

### CODE REVIEW

**Comment by michal-lightly** [comment]
@coderabbit full review

File: fast_track/src/bot/run-bot.test.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+interface FakeOptions {
+    existingApproval?: boolean;
+    associatedPullRequests?: unknown[];
+    // pulls.get gets called twice: once to reload the target before the approval,
+    // then once to re-check it afterwards. Configure the return values here.
+    reloadBeforeApprove?: PullRead;
+    recheckAfterApprove?: PullRead;
+}
+
+function fakeOctokit(options: FakeOptions = {}) {
```
[the remark this answers is not in the document]
Comment (michal-lightly) [reply fast_track/src/bot/run-bot.test.ts:46]: This is a self-contained test helper.

File: fast_track/src/bot/run-bot.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    };
+    return verdict.verdict === 'pass'
+        ? applyPass(params, target, verdict, actionParams)
+        : applyFailure(target, verdict, actionParams);
+}
+
+async function currentTarget(params: RunBotParams): Promise<BotTarget | null> {
+    const derivedTarget = await deriveTarget(params);
+    if (derivedTarget === null) return null;
+    return refreshTarget({ ...params, target: derivedTarget });
```
[the remark this answers is not in the document]
Comment (michal-lightly) [reply fast_track/src/bot/run-bot.ts:47]: Deferred to a follow-up PR.

File: fast_track/src/bot/run-bot.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ * caller fails closed. A throw here would otherwise skip the post-approval
+ * rollback and leave an unverified approval active.
+ */
+async function refreshOrNull(
+    params: Parameters<typeof refreshTarget>[0]
+): Promise<BotTarget | null> {
+    try {
+        return await refreshTarget(params);
+    } catch {
+        return null;
```
Comment (LeonardoRosaa) [inline fast_track/src/bot/run-bot.ts:65]: Should we at least log the error?

**Review by LeonardoRosaa — APPROVED** [review]

File: fast_track/src/bot/run-bot.ts (hunk 3 above)
Comment (michal-lightly) [reply fast_track/src/bot/run-bot.ts:65]: Good idea
